### PR TITLE
feat(storage): implement Redis bit operation commands (GETBIT, SETBIT, BITCOUNT, BITOP, BITPOS)  

### PR DESCRIPTION
**Pull request by lsyulong**
## Description  
Implemented Redis bit operation commands for string data type.
  
## Changes  
- Implement `setbit()` method: sets or clears the bit at offset in the string value stored at key  
- Implement `getbit()` method: returns the bit value at offset in the string value stored at key  
- Implement `bitcount()` method: counts the number of set bits in a string, with optional byte range support  
- Implement `bitop()` method: performs bitwise operations (AND, OR, XOR, NOT) between strings  
- Implement `bitpos()` method: finds the first bit set to 1 or 0 in a string, with three variants:  
  - `bitpos(key, bit)`: search entire string  
  - `bitpos_with_start(key, bit, start)`: search from start offset  
  - `bitpos_with_range(key, bit, start, end)`: search within byte range  
- Add `BitOpType` enum to define bitwise operation types  
- Export `BitOpType` in `src/storage/src/lib.rs` for public API access  
  
## Technical Details  
- Uses LSB-first bit ordering within bytes (bit 0 is least significant bit)  
- Supports negative offsets for Python-style indexing from string end  
- Handles expired keys correctly (returns 0 for GETBIT, appropriate values for BITPOS)  
- Implements proper locking with `ScopeRecordLock` for concurrent access  
- Stores results in `MetaCF` column family using `BaseKey` encoding  
- All operations follow existing code patterns in `src/storage/src/redis_strings.rs`  
  
## Testing  
- [✓] Compilation passes  
- [✓] Formatting checks pass  
- [✓] Clippy static analysis passes  
- [✓] Added comprehensive test suite in `src/storage/tests/redis_string_test.rs`:  
  - `test_setbit_getbit_basic`: basic set/get operations  
  - `test_setbit_getbit_multiple_offsets`: multiple bit positions  
  - `test_bitcount_basic`: bit counting without range  
  - `test_bitcount_with_range`: bit counting with byte ranges  
  - `test_bitop_and/or/xor/not`: all bitwise operations  
  - `test_bitpos_basic/with_start_offset/with_range`: all BITPOS variants  
  - `test_bitpos_not_found`: edge cases for missing bits  
  - `test_bit_operations_edge_cases`: invalid parameters  
  - `test_bit_operations_large_offsets`: large offset handling  
  
## Issue  
[#144 ]
  
## Summary by CodeRabbit  
**New Features**  
- Complete implementation of Redis bit manipulation commands (GETBIT, SETBIT, BITCOUNT, BITOP, BITPOS)  
- Support for all bitwise operations: AND, OR, XOR, NOT  
- Flexible bit position search with optional byte range parameters  
  
**Bug Fixes**  
- Consistent LSB-first bit ordering across all bit operations  
- Proper handling of expired keys and non-existent keys  
- Correct boundary condition handling for negative offsets  
  
**Tests**  
- Added 13 comprehensive test cases covering all bit operations  
- Tests for edge cases, large offsets, and concurrent access  
- Validation of bit ordering consistency across operations

## Summary by CodeRabbit

* **New Features**
  * Added bitwise storage operations: SETBIT, GETBIT, BITCOUNT, BITOP and BITPOS for byte/key-level bit manipulation.
  * Publicly exposed BitOpType (AND, OR, XOR, NOT) as part of the storage API.

* **Tests**
  * Added extensive tests validating setbit/getbit, bitcount, bitop, bitpos, ranges, edge cases, large offsets and concurrent scenarios.

### CODE REVIEW

File: src/storage/src/redis_strings.rs [hunk 1]
```diff
@@ -384,4 +391,610 @@ impl Redis {
...
+                    } else {
+                        // no range specified, count entire value
+                        (0, value_length.max(0) as usize)
+                    };
+
+                // get the slice and count bits
+                let slice = &data_value[start_byte..end_byte];
+                Ok(Self::get_bit_count(slice, slice.len() as i64))
+            }
+            None => Ok(0),
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:601]: 这里整个match的逻辑太长了，整个命令处理逻辑包含在一个{}里面，不方便阅读。

**Comment by guozhihao-224** [comment]
有一个统一的问题，String是写在Meta cf里面的，但是读的话，使用默认cf去读， 行为不一致

File: src/storage/src/redis_strings.rs [hunk 2]
```diff
@@ -384,4 +391,612 @@ impl Redis {
     //         }
     //     }
     // }
+    pub fn setbit(&self, key: &[u8], offset: i64, on: i32) -> Result<i32> {
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:394]: 这里多个命令都缺乏Type检测

File: src/storage/src/redis_strings.rs [hunk 3]
```diff
@@ -384,4 +391,612 @@ impl Redis {
...
+        // Validate offset
+        if offset < 0 || offset >= (1_i64 << 32) {
+            InvalidArgumentSnafu {
+                message: "offset must be 0 <= offset < 2^32".to_string(),
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:398]: 对于Redis逻辑上的命令问题，应该使用RedisErr返回

File: src/storage/src/redis_strings.rs [hunk 4]
```diff
@@ -384,4 +391,612 @@ impl Redis {
...
+                // get the slice and count bits
+                let slice = &data_value[start_byte..end_byte];
+                Ok(Self::get_bit_count(slice, slice.len() as i64))
+            }
+            None => Ok(0),
+        }
+    }
+
+    pub fn get_bit_count(value: &[u8], bytes: i64) -> i32 {
+        static BITS_IN_BYTE: [u8; 256] = [
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:606]: 将这个查表数组移到外面使用const定义，避免static开销

File: src/storage/src/redis_strings.rs [hunk 5]
```diff
@@ -384,4 +391,612 @@ impl Redis {
...
+                // Check if value is stale (expired)
+                if parsed_value.is_stale() {
+                    return Ok(0);
+                }
+
+                let data_value = parsed_value.user_value();
+
+                // Calculate byte and bit position (MSB-first)
+                let byte = (offset >> 3) as usize;
+                let bit = (7 - (offset & 0x7)) as usize;
```
Comment (guozhihao-224) [inline src/storage/src/redis_strings.rs:521]: 需要统一标注“mask”和被操作bit的直接的类型，统一在u8；整数默认i32，在做位移时，有可能存在潜在的坑

**Comment by lsyulong** [comment]
好的，我这边统一在调整下
